dump/parser.go

```diff
@@ -59,6 +64,25 @@ func Parse(r io.Reader, h ParseHandler, parseBinlogPos bool) error {
 			return c == '\r' || c == '\n'
 		})
 
+		if m := tableStartExp.FindAllStringSubmatch(line, -1); len(m) == 1 {
+			query = line
+			inTableParsing = true
+			continue
+		}
+
+		if inTableParsing {
+			if m := endExp.FindAllStringSubmatch(line, -1); len(m) == 1 {
```

Comment:
Why are you using a regex instead of simple `strings.IndexByte(line, ';')`?

Also, can you add some test cases (multi-line and one-line create table, ';' inside defaults as an example) for this new code? This change doesn't look intuitive and simple.

Reply:
resolved in [7344273](https://github.com/go-mysql-org/go-mysql/pull/767/commits/73442733bb257815fcf57c57a96c556e1054ea7e)